doc/developer/design/20220317_liveness_of_unmaterialized.md

```diff
@@ -0,0 +1,57 @@
+# Supporting liveness of weak reads on "unmaterialized" sources
+
+## Summary
+
+[This design document](https://github.com/MaterializeInc/materialize/pull/11262) documents
+how we want to achieve _strict serializability_ as an option for users querying sources
+in `materialized`. In addition, it mentions that currently, "materialized" sources are _serializable_ and "live".
+
+This document describes how we will add this liveness property to "unmaterialized" sources.
+
+## Goals
+
+Offer an eventually-consistent, live experience for "unmaterialized" sources, to match the semantics
+
+## Non-Goals
+
+Alter the default experience of using `materialized` (yet).
+
+## Description
+
+### Background
+
+This is in contrast to the current default for reads against "materialized" sources,
+which is `serializable` (we endeavor for completely consistent reads as any given timestamp),
+but not neccesarily _current_. We do, however, offer "liveness", which means
+we make sure that we are always making progress through the source's upstream.
+
+This is not a property we currently have for "unmaterialized" sources, which have no
+mechanism for ensuring liveness.
+
+We _do_ have a mechanism called _timestamp bindings_, which associated offsets and timestamps, which ensure _serializability_ for `materialized` sources, which have
+a default _index_ that continuously produces new bindings.
+
+### Solution
+
+Instead of creating a default _index_ on "unmaterialized" source creation, we can instead create a special source _instance_, that produces messages (with offsets),
+for which we store no actual data. This _instance_ will cause _timestamp bindings_ for these offets. When reads (also called "peeks") happen involving these
+sources, the pre-existing timestamp-selection logic will select compatible timestamps, causing us to present relatively up-to-date data.
+
+This will not affect any other query, even queries that force _strict serializability_. However, the alternative is true, and _strict serializable_ reads, as discussed
+the other design document, may cause these weaker reads to wait.
+
+### Platform considerations
+
+This plan should be compatible with being entirely contained within `STORAGE`, especially considering [this pr](https://github.com/MaterializeInc/materialize/pull/11223)
+
+## Alternatives
+
+- Change the default experience of using `materialized` to be `LINEARIZABLE`
+- Don't offer this experience, and force _strict serializability_ on uses of "unmaterialized" sources
+
+## Open questions
+
+- How much of a latency hit will fetching the max offset incur?
+- How feasible is it for ALL sources to offer a source instance only publishes timestamps?
+  - Can we reasonably not offer the above described experience to source types that can't offer the
```

Comment:
I think those sources are roughly Kinesis/Pubnub, which are both non-definite. And I think it's okay to force non-definite sources to be persisted, in which case they will have something that drives their timestamps forward!

Reply:
"force them to be persisted"

do you mean in the platform world?
I updated the doc, let me know what you think!